src/ld-query.js

```diff
@@ -345,32 +605,52 @@
 
     }
 
-    function QueryNode( jsonData ) {
+    function QueryNode( jsonData, parent ) {
 
         this.json = function() { return jsonData; };
+        var state = this._state = { cachePaths: true, paths: null };
+        if ( parent ) {
+            
+            var pstate = parent._state;
 
+            state.cachePaths = pstate.cachePaths;
+
+        }
     }
 
-    function buildQueryNode( jsonData ) { return new QueryNode( jsonData ); }
+    function buildQueryNode( parent ) {
 
-    QueryNode.prototype.query = function( path ) {
+        return function( json ) { return new QueryNode( json, parent ); }
 
-        // select the json targetted by this path
-        var selection = select( this.json(), path );
+    }
+
+    QueryNode.prototype.withPathCaching = function( cache ) {
+        cache = !!cache;
+        this._state.cachePaths = cache;
+        cache || ( this._state.paths = null );
+        return this;
+        
+    }
+
+    QueryNode.prototype.query = function( selector ) {
+
+        // select the json targetted by this selector
+        var selection = select( this._state, this.json(), selector );
```

Comment:
given that we're introducing the concept of cached queries, i'm thinking through whether it start to become a memory leak, but most use cases would seem to consist of creating the querynode, running one or more queries, then allowing the node to go out of scope, so probably this is not worth worrying about.

Ideally, if there was a way to overlay the caching functionality (by a decorator) on top of the underlying query object, users could elect to use it or not.